astropy/units/core.py

```diff
@@ -807,8 +807,18 @@ def __pow__(self, p: Real) -> CompositeUnit:
                 ) from None
             return CompositeUnit(1, [self], [sanitize_power(p)], _error_check=False)
 
+    @staticmethod
+    def _warn_about_operation_with_deprecated_type(op: str, other: bytes | str) -> None:
+        warnings.warn(
+            AstropyDeprecationWarning(
+                f"{op} involving a unit and a '{type(other).__name__}' instance are "
+                f"deprecated since v7.1. Convert {other!r} to a unit first."
+            )
```

Comment:
I think you want to set a proper `stacklevel` here. My guess is that it should be `stacklevel=3` in this instance.

Reply:
(setting a stacklevel) 👍. Specific level 🤷 I defer to others.